drivers/wifi/nrf_wifi/src/net_if.c

```diff
@@ -436,12 +439,20 @@ int nrf_wifi_if_send(const struct device *dev,
 						      nbuf);
 	} else {
 #endif /* CONFIG_NRF70_RAW_DATA_TX */
-		if ((vif_ctx_zep->if_carr_state != NRF_WIFI_FMAC_IF_CARR_STATE_ON) ||
-		    (!vif_ctx_zep->authorized && !is_eapol(pkt))) {
-			ret = -EPERM;
-			goto drop;
+
+		ra = nrf_wifi_util_get_ra(sys_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx], nbuf);
+		peer_id = nrf_wifi_fmac_peer_get_id(rpu_ctx_zep->rpu_ctx, ra);
+		if (peer_id == -1) {
+			nrf_wifi_osal_log_err("%s: Invalid peer",
```

Comment:
log `peer->ra_addr` for debugging. and do you really want a print in the data path? If you can use the Zephyr LOG macros you can use the ratelimited ones.